### PR TITLE
Plans: change default url of `BACK TO MY SITE` link

### LINKED ISSUE #5903: Plans: Plan purchase back button doesn't go to my-plan page
#### Steps to reproduce
1. You must purchase a plan through the NUX
2. On the thank you page, hit the "Back to my site" link
3. I see the /plans page

#### What I expected

I expected to see the "My Plan" page, which is actually kind of odd, because it replicates the "Thank you " page. I think these two things should be merged, and a thank you message added to the to of the my-plan page. That is probably a separate thing.

#### What happened instead

I saw the plans page, showing me the different plans, not the one I was on.

#### Browser / OS version

Chrome Mac latest

#### Screenshot / Video

This site, see the back link near the top:

screen shot 2016-06-08 at 1 58 51 pm

### PR DESCRIPTION
**Pull request by retrofox**
This PR fix #5903 modifying the default value of `<HeaderCake />` `BACK TO MY SITE` link.

The specific change is here -> b9aee1917f850a41aef2890037fd5770a95605bc. The other ones are simply eslint fixes.

### Testing

1. You must purchase a plan through the NUX. You should get a url similar to `http://calypso.localhost:3000/checkout/thank-you/<your-testing-new-site>/<receiptId>`
2. On the thank you page, hit the `BACK TO MY SITE` link
3. You should see `my-plan` instead of the plans page.

cc @apeatling 

Test live: https://calypso.live/?branch=update/checkout-thank-you-page

### CODE REVIEW

File: client/my-sites/upgrades/checkout-thank-you/index.jsx [hunk 1]
```diff
@@ -141,8 +141,12 @@ const CheckoutThankYou = React.createClass( {
 			const site = this.props.selectedSite.slug;
 
 			if ( purchases.some( isPlan ) ) {
-				page( `/plans/${ site }` );
-			} else if ( purchases.some( isDomainProduct ) || purchases.some( isDomainRedemption || purchases.some( isSiteRedirect ) ) ) {
+				page( `/plans/my-plan/${ site }` );
+			} else if (
+				purchases.some( isDomainProduct ) ||
+				purchases.some( isDomainRedemption ||
+				purchases.some( isSiteRedirect ) )
```
Comment (lamosty) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: On line 147, you are missing closing `)` right before `||` and on line 148 you have additional `)`. :)

Comment (lamosty) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: Oops, sorry. I just noticed that it's actually `purchases.some( isDomainRedemption || purchases.some( isSiteRedirect ) )`. Could you please leave this on one line? That's more readable.

**Comment by lamosty** [comment]
Working fine:

selection_027

Except for the non-blocking comments I left, this is ready to :ship:.

File: client/my-sites/upgrades/checkout-thank-you/index.jsx (hunk 1 above)
Comment (retrofox) [inline client/my-sites/upgrades/checkout-thank-you/index.jsx]: done! rebase in the last commit.

**Comment by retrofox** [comment]
Thanks @lamosty for the review.

**Pull request merged**